source/vibe/core/drivers/libasync.d

```diff
@@ -859,7 +859,7 @@ final class LibasyncManualEvent : ManualEvent {
 
 		bool signal_exists;
 
-		if (s_eventWaiters.length <= m_instance) {
+		if (s_eventWaiters.length <= m_instance + 1) {
```

Comment:
And this should cause `expandWaiters` to be called repeatedly for every `acquire` call on the latest `ManualEvent`.